lib/matplotlib/backends/backend_cairo.py

```diff
@@ -132,8 +132,18 @@ def _fill_and_stroke (self, ctx, fill_c, alpha, alpha_overrides):
             ctx.restore()
         ctx.stroke()
 
-    @staticmethod
-    def convert_path(ctx, path, transform):
+    def convert_path(self, ctx, path, transform):
+        # First clip the path to the actual rendering extents
+        points = path.get_extents().get_points()
+        width = points[1][0] - points[0][0]
+        height = points[1][1] - points[0][1]
+        wfactor = width / self.width
+        hfactor = height / self.height
```

Comment:
This is some logic to translate the cairo clip_extents to the actual path extents.

Let's say that
1) we draw on a canvas that is 1000 pixels wide (self.width)
2) The paths we're drawing have values ranging from 0 to 100000 "data units"
3) the region that we have to draw on (ctx.clip_extents()) is from 0 *pixels* to 100 *pixels*
4) We thus want to clip the path to 0 to 10000 "data units", ie region_width times "data width" divided by self.width (100 * 100000 / 1000)

I hope I've been clear enough, there may be other ways to achieve this but I am quite sure this one is correct now (compared to the previous proposal).